src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -50,6 +62,98 @@
     }
   }
 
+  /**
+   * Gets all of the characters in a string that are letters.
+   * @param str to get letters from.
+   * @return letter characters from string.
+   */
+  private String getAlphabeticChars(String str) {
+    StringBuilder sb = new StringBuilder();
+    for (char c : str.toCharArray()) {
+      if (Character.isAlphabetic(c)) {
+        sb.append(c);
+      }
+    }
+    String alphaChars = sb.toString();
+    return alphaChars;
+  }
+
+  /**
+   * Gets all of the digits from a string and forms and integer.
+   * @param str to get digits from.
+   * @return integer formed from string.
+   */
+  private int getIntFromString(String str) {
+    StringBuilder sb = new StringBuilder();
+    boolean hasNum = false;
+    for (char c : str.toCharArray()) {
+      if (Character.isDigit(c)) {
+        sb.append(c);
+        hasNum = true;
+      }
+    }
+    int num = 0;
+    if (hasNum) {
+      num = Integer.parseInt(sb.toString());
+    }
+    return num;
+  }
+
+  /**
+   * Formats date string of type "YYYYMMDD" to "YYYY-MM-DD".
+   * @param date string formatted as "YYYYMMDD".
+   * @return string of date in form "YYYY-MM-DD".
+   */
+  private String dateFormat(String date) {
```

Comment:
Try to avoid direct date manipulation. Generally, parsing and then formatting is safer